pkg/collector/corechecks/system/wincrashdetect/wincrashdetect.go

```diff
@@ -0,0 +1,200 @@
+	"github.com/DataDog/datadog-agent/pkg/util/retry"
+	"golang.org/x/sys/windows/registry"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const (
+	crashDetectCheckName = "wincrashdetect"
+	maxStartupWarnings   = 20
+)
+
+// Config is the configuration options for this check
+// it is exported so that the yaml parser can read it.
+type WinCrashConfig struct {
+	Enabled bool `yaml:"enabled"` // placeholder for config
+}
+
+type WinCrashDetect struct {
+	core.CheckBase
+	instance         *WinCrashConfig
+	hasRunOnce       bool
+	startupWarnCount int
+}
+
+func init() {
+	core.RegisterCheck(crashDetectCheckName, crashDetectFactory)
+}
+
+func crashDetectFactory() check.Check {
+	return &WinCrashDetect{
+		CheckBase:  core.NewCheckBase(crashDetectCheckName),
+		instance:   &WinCrashConfig{},
+		hasRunOnce: false,
+	}
+}
+
+// Parse parses the check configuration
+func (c *WinCrashConfig) Parse(data []byte) error {
+	// default values
+	c.Enabled = true
+
+	return yaml.Unmarshal(data, c)
+}
+
+func (wcd *WinCrashDetect) Configure(integrationConfigDigest uint64, config, initConfig integration.Data, source string) error {
+	err := wcd.CommonConfigure(integrationConfigDigest, initConfig, config, source)
+	if err != nil {
+		return err
+	}
+
+	return wcd.instance.Parse(config)
+}
+
+func (wcd *WinCrashDetect) handleStartupError(err error) error {
+	if retry.IsErrWillRetry(err) {
+		wcd.startupWarnCount++
+		// this is an expected error, the check run will occur before the system probe
+		// comes up.  However, it should resolve.  If the number of these exceeds a
+		// given threshold, report as an error anyway
+		if wcd.startupWarnCount > maxStartupWarnings {
+			return err
+		}
+		return nil
+	}
+	return err
+}
+func (wcd *WinCrashDetect) Run() error {
+
+	// only do this once; there's no point in doing it after the first one
+	if wcd.hasRunOnce {
+		log.Infof("check already run")
+		return nil
+	}
+	sysProbeUtil, err := process_net.GetRemoteSystemProbeUtil(
+		dd_config.SystemProbe.GetString("system_probe_config.sysprobe_socket"))
+	if err != nil {
+		return wcd.handleStartupError(err)
+	}
+
+	data, err := sysProbeUtil.GetCheck(sysconfig.WindowsCrashDetectModule)
+	if err != nil {
+		return wcd.handleStartupError(err)
+	}
+	crash, ok := data.(probe.WinCrashStatus)
+	if !ok {
+		return fmt.Errorf("Raw data has incorrect type")
+	}
+	/*
+	 * originally did this with a sync.once.  The problem is the check is run prior to the
+	 * system probe being successfully started.  This is OK; we just need to detect the BSOD
+	 * on first available run.
+	 *
+	 * so keep running the check until we are able to talk to system probe, after which
+	 * we don't need to run any more
+	 */
+	wcd.hasRunOnce = true
+	if !crash.Success {
+		return fmt.Errorf("Error getting crash data %s", crash.ErrString)
+	}
+
+	if len(crash.FileName) == 0 {
+		// no crash data present.  this is actually good news.  We don't need to do anything
+		// else
+		return nil
+	}
+
+	if haveAlreadyReported(crash) {
```

Comment:
Would not be simpler to save a timestamp and collect dump/dumps information which happen after the time stamp to avoid loading and analyzing the same dump.

Reply:
again, this is designed to be used by multiple consumers.

Reply:
Not sure I understand what multiple consumers you are referring. End even if there are multiple consumers single timestamp bookmark would work as well. I it is just a suggestion though.